database/database.go

```diff
@@ -16,20 +16,21 @@ type DB struct {
 }
 
 type Scan struct {
-	ID               int64             `json:"id"`
-	Timestamp        time.Time         `json:"timestamp"`
-	Target           string            `json:"target"`
-	Replay           int               `json:"replay"` //hours or days
-	Has_tls          bool              `json:"has_tls"`
-	Cert_id          int64             `json:"cert_id"`
-	Trust_id         int64             `json:"trust_id"`
-	Is_valid         bool              `json:"is_valid"`
-	Validation_error string            `json:"validation_error,omitempty"`
-	Complperc        int               `json:"completion_perc"`
-	Conn_info        connection.Stored `json:"connection_info"`
-	AnalysisResults  []Analysis        `json:"analysis,omitempty"`
-	Ack              bool              `json:"ack"`
-	Attempts         int               `json:"attempts"` //number of retries
+	ID               int64                  `json:"id"`
+	Timestamp        time.Time              `json:"timestamp"`
+	Target           string                 `json:"target"`
+	Replay           int                    `json:"replay"` //hours or days
+	Has_tls          bool                   `json:"has_tls"`
+	Cert_id          int64                  `json:"cert_id"`
+	Trust_id         int64                  `json:"trust_id"`
+	Is_valid         bool                   `json:"is_valid"`
+	Validation_error string                 `json:"validation_error,omitempty"`
+	Complperc        int                    `json:"completion_perc"`
+	Conn_info        connection.Stored      `json:"connection_info"`
+	AnalysisResults  []Analysis             `json:"analysis,omitempty"`
+	Ack              bool                   `json:"ack"`
+	Attempts         int                    `json:"attempts"` //number of retries
+	AnalysisParams   map[string]interface{} `json:"analysis_params"`
```

Comment:
This is a nitpick, and I'm fine leaving this as a `map[string]interface{}`, but for the sake of the discussion I generally try to avoid `map[string]interface{}` as much as possible. `AnalysisResults`, for example, uses an `[]Analysis` where the name of the analysis worker is included in the `Analysis` struct.
```go
type Analysis struct {
        ID       int64           `json:"id"`
        Analyzer string          `json:"analyzer"`
        Result   json.RawMessage `json:"result"`
}
```
The upside of this approach is the analyzer name is typed. Downside: you need to iterate through the array to find the entry that applies to a given worker, instead of doing a hash lookup. In this case, the performance cost is irrelevant given we'll never have more than a couple entries in the array.